### ISSUE TITLE
Deployments not deleted after finishes

### ISSUE THREAD

**Issue by dwctua** [GoogleCloudPlatform/cloud-code-intellij#2801]
the 'delete deployments' checkbox is checked.
![image](https://user-images.githubusercontent.com/46290157/88964938-0d57fc80-d278-11ea-9126-9033539a983b.png)
### Expected Behavior
After hitting 'stop', should see the deployments are deleted and the cleaning up in logs
```
Cleaning up...
 - deployment.apps "java-hello-world" deleted
 - service "java-hello-world-external" deleted
```

### Actual Behavior
After hitting 'stop', the deployment is still running in the cluster.

### Additional Information

- IDE type: IntelliJ community
- IDE version: 2020.2
- Cloud Code version: 20.6.1-202
- Cloud SDK (Settings > Cloud Code > Cloud SDK)
  - Are you allowing the plugin to manage the Cloud SDK: yes
  - Version of the Cloud SDK: 302.0.0
- Skaffold version (If you are using Kubernetes features; Settings > Cloud Code > Kubernetes): v1.10.1
- Operating System: Windows 10

_Feel free to deviate from this template as needed, especially if you are submitting a feature request._

**Comment by etanshaul** [comment]
thanks @dwctua 

would you mind sharing the output logs from the deployment session? And if you could turn on `DEBUG` level verbosity too that would help:
<img width="717" alt="Screen Shot 2020-07-30 at 3 35 55 PM" src="https://user-images.githubusercontent.com/1735744/88966488-632da400-d27a-11ea-8119-12dafa3e21ef.png">

**Comment by dwctua** [comment]
actually I turned on trace, nothing is printed.

**Comment by etanshaul** [comment]
@dwctua just to be sure, this is the output window I meant:

<img width="665" alt="Screen Shot 2020-07-30 at 4 09 19 PM" src="https://user-images.githubusercontent.com/1735744/88969575-10a2b680-d27f-11ea-8585-b6a7dd7ed244.png">

Does this panel show up empty (or not appear) for you?

**Comment by dwctua** [comment]
[java-hello-world-6d9586b757-6lvgb server] Picked up JAVA_TOOL_OPTIONS: -agentlib:jdwp=transport=dt_socket,server=y,address=5005,suspend=n,quiet=y
[java-hello-world-6d9586b757-6lvgb server] 
[java-hello-world-6d9586b757-6lvgb server]   .   ____          _            __ _ _
[java-hello-world-6d9586b757-6lvgb server]  /\\ / ___'_ __ _ _(_)_ __  __ _ \ \ \ \
[java-hello-world-6d9586b757-6lvgb server] ( ( )\___ | '_ | '_| | '_ \/ _` | \ \ \ \
[java-hello-world-6d9586b757-6lvgb server]  \\/  ___)| |_)| | | | | || (_| |  ) ) ) )
[java-hello-world-6d9586b757-6lvgb server]   '  |____| .__|_| |_|_| |_\__, | / / / /
[java-hello-world-6d9586b757-6lvgb server]  =========|_|==============|___/=/_/_/_/
[java-hello-world-6d9586b757-6lvgb server]  :: Spring Boot ::        (v2.1.4.RELEASE)
[java-hello-world-6d9586b757-6lvgb server] 
[java-hello-world-6d9586b757-6lvgb server] 2020-07-30 20:25:13.238  INFO 1 --- [           main] c.helloworld.HelloWorldApplication       : Starting HelloWorldApplication on java-hello-world-6d9586b757-6lvgb with PID 1 (/app/classes started by 1001580000 in /)
[java-hello-world-6d9586b757-6lvgb server] 2020-07-30 20:25:13.241  INFO 1 --- [           main] c.helloworld.HelloWorldApplication       : No active profile set, falling back to default profiles: default
time="2020-07-30T16:25:13-04:00" level=debug msg="port forwarding service-java-hello-world-external-odo-test-80 got terminated: exit status 1, output: error: error upgrading connection: unable to upgrade connection: pod not found (\"java-hello-world-6f8bd955c6-wsvg4_odo-test\")\n"
time="2020-07-30T16:25:14-04:00" level=debug msg="Using wrapper mvnw.cmd for mvn"
time="2020-07-30T16:25:14-04:00" level=debug msg="could not stat dependency: CreateFile C:\\\\Ming\\\\projects\\\\cloud-code-samples\\\\java\\\\java-hello-world\\\\src\\\\main\\\\jib: The system cannot find the file specified."
time="2020-07-30T16:25:14-04:00" level=debug msg="Found dependencies for jib maven artifact: [pom.xml src\\main\\java\\cloudcode\\helloworld\\HelloWorldApplication.java src\\main\\java\\cloudcode\\helloworld\\web\\HelloWorldController.java src\\main\\resources\\application.properties src\\main\\resources\\application.properties]"
time="2020-07-30T16:25:14-04:00" level=debug msg="Running command: [kubectl --context odo-test/api-us-east-2-starter-openshift-online-com:6443/dwctua@gmail.com --namespace odo-test port-forward --pod-running-timeout 1s service/java-hello-world-external 80:80 --namespace odo-test]"
[java-hello-world-6d9586b757-6lvgb server] 2020-07-30 20:25:15.403  INFO 1 --- [           main] org.eclipse.jetty.util.log               : Logging initialized @3466ms to org.eclipse.jetty.util.log.Slf4jLog
[java-hello-world-6d9586b757-6lvgb server] 2020-07-30 20:25:15.607  INFO 1 --- [           main] o.s.b.w.e.j.JettyServletWebServerFactory : Server initialized with port: 8080
[java-hello-world-6d9586b757-6lvgb server] 2020-07-30 20:25:15.610  INFO 1 --- [           main] org.eclipse.jetty.server.Server          : jetty-9.4.15.v20190215; built: 2019-02-15T16:53:49.381Z; git: eb70b240169fcf1abbd86af36482d1c49826fa0b; jvm 1.8.0_252-8u252-b09-1~deb9u1-b09
[java-hello-world-6d9586b757-6lvgb server] 2020-07-30 20:25:15.697  INFO 1 --- [           main] o.e.j.s.h.ContextHandler.application     : Initializing Spring embedded WebApplicationContext
[java-hello-world-6d9586b757-6lvgb server] 2020-07-30 20:25:15.698  INFO 1 --- [           main] o.s.web.context.ContextLoader            : Root WebApplicationContext: initialization completed in 2380 ms
[java-hello-world-6d9586b757-6lvgb server] 2020-07-30 20:25:15.996  INFO 1 --- [           main] org.eclipse.jetty.server.session         : DefaultSessionIdManager workerName=node0
[java-hello-world-6d9586b757-6lvgb server] 2020-07-30 20:25:15.996  INFO 1 --- [           main] org.eclipse.jetty.server.session         : No SessionScavenger set, using defaults
[java-hello-world-6d9586b757-6lvgb server] 2020-07-30 20:25:15.997  INFO 1 --- [           main] org.eclipse.jetty.server.session         : node0 Scavenging every 600000ms
[java-hello-world-6d9586b757-6lvgb server] 2020-07-30 20:25:16.012  INFO 1 --- [           main] o.e.jetty.server.handler.ContextHandler  : Started o.s.b.w.e.j.JettyEmbeddedWebAppContext@6869a3b3{application,/,[file:///tmp/jetty-docbase.8008148518379525325.8080/],AVAILABLE}
[java-hello-world-6d9586b757-6lvgb server] 2020-07-30 20:25:16.013  INFO 1 --- [           main] org.eclipse.jetty.server.Server          : Started @4077ms
[java-hello-world-6d9586b757-6lvgb server] 2020-07-30 20:25:16.309  INFO 1 --- [           main] o.s.s.concurrent.ThreadPoolTaskExecutor  : Initializing ExecutorService 'applicationTaskExecutor'
[java-hello-world-6d9586b757-6lvgb server] 2020-07-30 20:25:16.881  INFO 1 --- [           main] o.e.j.s.h.ContextHandler.application     : Initializing Spring DispatcherServlet 'dispatcherServlet'
[java-hello-world-6d9586b757-6lvgb server] 2020-07-30 20:25:16.881  INFO 1 --- [           main] o.s.web.servlet.DispatcherServlet        : Initializing Servlet 'dispatcherServlet'
[java-hello-world-6d9586b757-6lvgb server] 2020-07-30 20:25:16.887  INFO 1 --- [           main] o.s.web.servlet.DispatcherServlet        : Completed initialization in 6 ms
[java-hello-world-6d9586b757-6lvgb server] 2020-07-30 20:25:16.908  INFO 1 --- [           main] o.e.jetty.server.AbstractConnector       : Started ServerConnector@522bf64e{HTTP/1.1,[http/1.1]}{0.0.0.0:8080}
[java-hello-world-6d9586b757-6lvgb server] 2020-07-30 20:25:16.911  INFO 1 --- [           main] o.s.b.web.embedded.jetty.JettyWebServer  : Jetty started on port(s) 8080 (http/1.1) with context path '/'
[java-hello-world-6d9586b757-6lvgb server] 2020-07-30 20:25:16.914  INFO 1 --- [           main] c.helloworld.HelloWorldApplication       : Started HelloWorldApplication in 4.412 seconds (JVM running for 4.978)

**Comment by etanshaul** [comment]
great! those logs look good so far. The application is still running here. 

Could you also hit the stop button (the one on the left in my screenshot above) and share the extra logs that come up? After hitting that button, I'd expect your deployment to be cleaned up.

**Comment by dwctua** [comment]
no more output, this is after hit the stop button.

**Comment by etanshaul** [comment]
oh ok. interesting. that sounds like a bug, but I am not yet able to reproduce. I will keep investigating.

If you can share - what kind of cluster are you deploying to? (e.g. GKE, Minikube, Docker Desktop..)

**Comment by dwctua** [comment]
openshift

**Comment by dwctua** [comment]
I switched to my local k8s (Docker-desktop) and still the same, but getting one extra message:
time="2020-07-30T17:33:06-04:00" level=debug msg="port forwarding service-java-hello-world-external-default-80 got terminated: exit status 4294967295, output: "

**Comment by etanshaul** [comment]
bringing in some skaffold expertise: @briandealwis @nkubala

This is using skaffold `1.10.1` (which will update to the latest in the next Cloud Code release). Is there a scenario that you can think of that would cause deployments not to cleanup properly? I wasn't immediately able to repro myself using matching versions.

**Comment by briandealwis** [comment]
Sounds like GoogleContainerTools/skaffold#4058 — we haven't been able to reproduce, though we have a number of confirmations on the issue.

**Comment by dwctua** [comment]
In my case, it is slightly different. Both CTL+C or command line via 'skaffold delete' is working for me.

**Comment by etanshaul** [comment]
> In my case, it is slightly different. Either CTL+C or command line via 'skaffold delete' is working for me.

Interesting. I wonder if it is terminal-dependent somehow. Now that we have narrowed it down to (probably) a Windows thing, I will try and repro on that platform.

**Comment by etanshaul** [comment]
Kamino cloned this issue to [GoogleCloudPlatform/cloud-code-intellij-internal](https://github.com/GoogleCloudPlatform/cloud-code-intellij-internal/issues/2268)

**Comment by etanshaul** [comment]
@dwctua thank you for bringing this to our attention. We have identified and fixed the and it will be available in the next release.

**Comment by etanshaul** [comment]
this should now be fixed with the release of `20.8.1`. Let us know if you run into any issues. Thanks.
